src/utility/Auth.js

```diff
@@ -0,0 +1,55 @@
+import React, { createContext, useContext, useState, useEffect } from 'react'
+import firebase from 'firebase/app'
+import 'firebase/auth'
+
+const AuthContext = createContext()
+
+export function useAuth() {
+  return useContext(AuthContext)
+}
+
+export function AuthProvider({ children }) {
+  const [user, setUser] = useState(null)
+
+  useEffect(() => {
+    return firebase.auth().onAuthStateChanged(async currUser => {
+      setUser(currUser)
+    })
+  })
+
+  return (
+    <AuthContext.Provider value={{ isAuthed: !!user, user, setUser }}>
+      {children}
+    </AuthContext.Provider>
+  )
+}
+
+export const googleSignIn = async (setUser, setLocation) => {
+  await firebase.auth().setPersistence(firebase.auth.Auth.Persistence.LOCAL)
+  const provider = new firebase.auth.GoogleAuthProvider()
+  try {
+    await firebase.auth().signInWithPopup(provider)
+    const user = firebase.auth().currentUser
+    setUser(user)
+    setLocation('/application')
+  } catch (e) {
+    // Not sure what to do here.
```

Comment:
might be good to multiplex on the error here to provide more info the user, can probably use `TextInputLikeErrorMsg` from `Common.js`

Reply:
I don't quite understand what you mean, could you go into more detail?

Reply:
It would be nice to display an error, maybe underneath the login buttons that lists the error. I'm trying to login on the preview link and its not working andnot seeing any errors is weird

Reply:
I'll add a banner that pops up when the login fails.

Regarding logging in through a preview link: unfortunately its not really possible. Github only allows us to specify one link per app, and firebase only lets us select one app per firebase project. Currently the specified link for the dev firebase is localhost so it can be tested locally, do you think I should change it to amplify?

Reply:
(Also, we need to specify the exact url, including subdomains, for github. so it probably won't like the way amplify makes our previews)

Reply:
I've added an error banner.